docs/plugins/reindex.asciidoc

```diff
@@ -0,0 +1,608 @@
+[[plugins-reindex]]
+=== Reindex Plugin
+
+The reindex plugin adds two APIs:
+
+* `_update_by_query` updates all documents matching a query in place.
+* `_reindex` copies documents from one index to another.
+
+These APIs are brothers so they live in the same plugin. Both use
+{ref}/search-request-scroll.html[Scroll] and {ref}/docs-bulk.html[Bulk] APIs
+to send an index request per document though they may not work that way in the
+future. Regardless of how they work internally they will always work **as**
+**though** they send an index request per document that matches a scroll.
+
+[float]
+==== Installation
+
+This plugin can be installed using the plugin manager:
+
+[source,sh]
+----------------------------------------------------------------
+sudo bin/plugin install reindex
+----------------------------------------------------------------
+
+The plugin must be installed on every node in the cluster, and each node must
+be restarted after installation.
+
+[float]
+==== Removal
+
+The plugin can be removed with the following command:
+
+[source,sh]
+----------------------------------------------------------------
+sudo bin/plugin remove reindex
+----------------------------------------------------------------
+
+The node must be stopped before removing the plugin.
+
+[[update-by-query-usage]]
+==== Using `_update_by_query`
+
+The simplest usage of `_update_by_query` just `touch`es every document in an
+index:
+
+[source,js]
+--------------------------------------------------
+POST /twitter/_update_by_query
+--------------------------------------------------
+// AUTOSENSE
+
+That will return something like this:
+
+[source,js]
+--------------------------------------------------
+{
+  "took" : 639,
+  "updated": 1235,
+  "batches": 13,
+  "version_conflicts": 2,
+  "failures" : [ ]
+}
+--------------------------------------------------
+
+`_update_by_query` gets a snapshot of the index when it starts and indexes what
+it finds using `internal` versioning. That means that if the document changes
+between the time when the snapshot was taken and when the index request is
+processed then you'll get a version conflict. When the versions match the
```

Comment:
I would turn the sentence around: That means that you'll get a version conflict if the document changes after the snapshot was taken and before the index request was processed.